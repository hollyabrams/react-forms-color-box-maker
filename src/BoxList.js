import { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

function BoxList() {
  const [boxes, setBoxes] = useState([]);
  const [boxId, setBoxId] = useState(0);

  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: boxId }]);
    setBoxId((id) => id + 1);
  };

  const removeBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  return (
    <div>
      <NewBoxForm addBox={addBox} />
      {boxes.map((box) => (
        <Box
          key={box.id}
          id={box.id}
          width={box.width}
          height={box.height}
          backgroundColor={box.backgroundColor}
          removeBox={removeBox}
        />
      ))}
    </div>
  );
}

export default BoxList;

