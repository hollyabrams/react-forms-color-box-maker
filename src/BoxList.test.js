import { render } from '@testing-library/react';
import BoxList from './BoxList';

describe('BoxList', () => {
  it('renders without crashing', () => {
    render(<BoxList />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
