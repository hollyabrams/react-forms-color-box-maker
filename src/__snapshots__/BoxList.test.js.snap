// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxList matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="widthInput"
      >
        Width:
      </label>
      <input
        id="widthInput"
        name="width"
        type="number"
        value=""
      />
      <label
        for="heightInput"
      >
        Height:
      </label>
      <input
        id="heightInput"
        name="height"
        type="number"
        value=""
      />
      <label
        for="backgroundColorInput"
      >
        Background Color:
      </label>
      <input
        id="backgroundColorInput"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
